Source/MLX/Nested.swift

```diff
@@ -350,6 +350,48 @@ public indirect enum NestedItem<Key: Hashable, Element>: IndentedDescription {
         }
     }
 
+    /// A combination of ``mapValues(_:)``, ``flattened(prefix:)`` and ``unflattened(_:)``.
+    ///
+    /// This transforms a nested structure into a new nested structure with the same shape but transformed
+    /// values.  The `transform` function receives a dotted path and the value to transform.
+    public func mapValues<Result>(
```

Comment:
This is the equivalent of `tree_map_with_path` ?

Reply:
yes!